File: src/samsung/Amplification.java
package samsung;

import java.util.Scanner;

public class Amplification {

	public static void main(String[] args) {
		int n=3,m=3;
		int mat[][] = {
				{1,0,0},
				{1,0,1},
				{1,0,0}
		};
//		int k=2,col=0;
//		System.out.println(solve(mat,col,k,n,m));
	}
	void change(int m[][]) {
		
	}
	
	private static int solve(int mat[][],int col,int k,int n,int m) {
		int result = 0;
		if(k%2==0) {
			int count=0;
			int j=0;
			for(int i=0;i<n;i++) {
				for(j=0;j<m;j++) {
					if(mat[i][j]==0) {
						break;
					}
					if(j==m) {
						count++;
					}
				}
			}
			return count;
		}
		if(col==m || k==0) {
			return 0;
		}
		toggleCol(mat,col);
		result = Math.max(result,solve(mat,col+1,k-1,n,m));
		toggleCol(mat,col);
		result = Math.max(result,solve(mat,col+1,k,n,m));
		return result;
	}//solve
	
	
	private static void toggleCol(int mat[][],int col) {
		for(int i=0;i<mat.length;i++) {
			if(mat[i][col]==1) {
				mat[i][col]=0;
			}else {
				mat[i][col]=1;
			}
		}
	}//toggle
	
	static void print(int [][]a) {
		System.out.println("++++++");
		for(int r[]:a) {
			for(int j:r) {
				System.out.print(j+" ");
			}
			System.out.println();
		}
		System.out.println("++++++");
	}
	
}

File: bin/samsung/final/1_Amplification/Amplification.cpp
#include<iostream>
using namespace std;
int mat[100][20];
int n,m;

void toggle(int col)
{
	for(int i=0;i<n;i++)
	{
		if(mat[i][col]==1)
			mat[i][col]=0;
		else
			mat[i][col]=1;
	}
}


int func(int col,int k,int &ans)
{
	int result = 0;
	if(k%2==0)
	{
		int i,j,count=0;
		for(i=0;i<n;i++)
		{
			for(j=0;j<m;j++)
			{
				if(mat[i][j]!=1)
					break;
			}
			if(j==m)
				count++;
		}
		if(count>ans)	
		    ans=count;
		  result = count;
	}
	if(col==m||k==0)
		return 0;
	toggle(col);
	result = max(result,func(col+1,k-1,ans));
	toggle(col);
	result = max(result,func(col+1,k,ans));
	return result;
}

int main()
{
	int t;
	cin>>t;
	while(t--)
	{
	    int k;
	    int ans=0;
		cin>>n>>m>>k;
		for(int i=0;i<n;i++)
			for(int j=0;j<m;j++)
				cin>>mat[i][j];
		func(0,k,ans);
		cout<<ans<<endl;	
	}
	return 0;
}
